File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.cpp
class Solution {
public:
    int findMin(vector<int> &nums) {
        //Compare mid with right to converge upon minimum element
        int left = 0;
        int right = nums.size()-1;
        while(left < right){
            int mid = (left+right)/2;
            if(nums[mid] > nums[right]){
                left = mid+1;
            } else if(nums[mid] <= nums[right]){
                right = mid;
            }
        }
        return nums[right];
    }
};

File: Data Structures & Algorithms/find-duplicate-integer/submission-0.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        count = {}
        for i in nums:
            if i not in count:
                count[i] = 0
            count[i]+=1
        for j in count:
            if count[j] > 1:
                return j
        